paper/src/main/java/vg/civcraft/mc/civmodcore/world/locations/chunkmeta/WorldChunkMetaManager.java

```diff
@@ -111,20 +112,17 @@ private ChunkCoord getChunkCoord(int x, int z, boolean gen, boolean populate) {
 			if (populate) {
 				// up until here we are still sync from the ChunkLoadEvent, so we need to
 				// offload the actual db load to another thread
-				synchronized (chunkLoadingQueue) {
-					chunkLoadingQueue.add(coord);
-					chunkLoadingQueue.notifyAll();
-				}
+				chunkLoadingExecutor.execute(coord::loadAll);
```

Comment:
This is a bit dangerous without additional synchronization. The pool may delay work until the year 3000, but access to the object isn't blocked via futures until then. This needs some form of blocking until the operation (loadAll) is really done.

Reply:
No it doesn't, ChunkCoord#getMeta will block if necessary, otherwise this is blocking on a main thread

Reply:
But this is still going to block when no more db connections are available and just stall another thread, that in turn will never notify the main thread when it runs into ChunkCoord#getMeta. I still think this isn't the solution but a bandaid, that may just delay the failure where no thread wakes up in getMeta.